XenAdmin/Commands/InstallCertificateCommand.cs

```diff
@@ -0,0 +1,112 @@
+﻿/* Copyright (c) Citrix Systems, Inc. 
+ * All rights reserved. 
+ * 
+ * Redistribution and use in source and binary forms, 
+ * with or without modification, are permitted provided 
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using XenAdmin.Core;
+using XenAdmin.Dialogs;
+using XenAPI;
+
+namespace XenAdmin.Commands
+{
+    class InstallCertificateCommand : Command
+    {
+        public InstallCertificateCommand()
+        {
+        }
+
+        public InstallCertificateCommand(IMainWindow mainWindow, Host host)
+            : base(mainWindow, host)
+        {
+        }
+
+        public InstallCertificateCommand(IMainWindow mainWindow, IEnumerable<SelectedItem> selection)
+            : base(mainWindow, selection)
+        {
+        }
+
+        protected override void ExecuteCore(SelectedItemCollection selection)
+        {
+            var host = selection.AsXenObjects<Host>().First();

```

Comment:
If there are no suitable elements found this could crash out.
Assuming this can happen in real world usage, I'd recommend this pattern:
```c#
var host = selection.AsXenObjects<Host>().FirstOrDefault();
if (host == null)
    return;
```

Reply:
I see, @kc284 this case would be prevented by `CanExecuteCore`.